### ISSUE TITLE
Feature request: gif output for Wesnoth's `--render-image` flag

### ISSUE THREAD

**Issue by cooljeanius** [wesnoth/wesnoth#7386]
### Describe the desired feature

Sometimes I would like to see how an animation would look without starting the entire game. I see from `--help` output that Wesnoth has a `--render-image` flag to process IPFs into PNGs, so I was thinking that it would be nice if that flag could be extended to support the progressive string square bracket expansion found in AnimationWML, so that it could be used to produce animations as well. If adding an additional dependency for gif support would be prohibitive, I guess some other animation format (such as APNG) could work, too.

**Comment by CelticMinstrel** [comment]
While the idea of rendering a progressive string is a good one, it's probably easier to make it output multiple images rather than a single animated image.

**Comment by cooljeanius** [comment]
I mean, part of the idea is that the rendered animation would take into account the same timing information that goes into displaying the animation in-game, and I don't think there's a way to convey timing information solely via a series of static images...

**Comment by CelticMinstrel** [comment]
There totally is, though it's not exactly space-efficient – just produce duplicate images as necessary so that each of the static images is meant to show for the exact same period of time.

I do see your point, but this way can at least be done without pulling in a library for a specific format. I also wouldn't be surprised if there's some standard way to output the timing as an auxiliary text file (something like a .cue file; not sure .cue specifically can do it, though). Maybe SVG can do it, though that seems like the wrong tool for the job.

**Comment by cooljeanius** [comment]
I guess another idea could be to put the timing info in the file names? e.g.,
`frames001-125.png`, `frames126-150.png`, and so on

**Comment by CelticMinstrel** [comment]
That's true. Or output a file similar to what's described [here}(https://video.stackexchange.com/a/28177). Or both. The main point being that providing all the information to allow dedicated tools to make a video is probably going to be _much_ easier than actually making the video.

**Comment by cooljeanius** [comment]
> That's true. Or output a file similar to what's described [here](https://video.stackexchange.com/a/28177). Or both. The main point being that providing all the information to allow external tools to make a video is probably going to be _much_ easier than actually making the video.
> 
> Incidentally, if outputting a file as described above, I bet you could actually pipe Wesnoth's output straight into ffmpeg.

We'd probably want to read up on `ffmpeg`'s documentation first to make sure...

**Comment by cooljeanius** [comment]
I guess if #9725 gets fixed, we could use WebP/WebM support instead...
